packages/manager/modules/emailpro/src/domain/dkim-autoconfig/dkim-auto-configurator.js

```diff
@@ -0,0 +1,140 @@
+    this.selector2NameInfos = this.getDkimName(2);
+    this.selector2RecordInfos = this.getDkimRecord(2);
+  }
+
+  async loadDataForDkim() {
+    this.showConfiguratingBtn = false;
+    this.showSpfDiagnosticTitle = false;
+    this.loading = true;
+    this.dkimSelectorsNoDomain = await this.getDkimSelectorForCurrentState();
+    await this.stepConfigureDkimFor('emailpro');
+    this.isStepConfigureValid = true;
+    this.loading = false;
+  }
+
+  getDkimSelector() {
+    return this.getDkimSelectorForCurrentState().then((dkimSelectors) => {
+      this.dkimSelectorsNoDomain = dkimSelectors;
+    });
+  }
+
+  leaveDkimConfigurator(reload = false) {
+    this.services.navigation.resetAction();
+    this.initializeDkimConfiguratorNoOvh();
+    this.services.$state.go(
+      'email-pro.dashboard.domain',
+      {},
+      reload ? { reload: 'email-pro.dashboard.domain' } : null,
+    );
+  }
+
+  stepConfigureDkimFor(targetService) {
+    const promises = this.postDkimFor(this.dkimSelectorsNoDomain);
+    return this.services.$q
+      .all(promises)
+      .then(() => {
+        this.writeSuccess(
+          `${targetService}_tab_domain_diagnostic_dkim_activation_success`,
+        );
+      })
+      .catch(() => {
+        this.leaveDkimConfigurator();
+        this.writeError(`${targetService}_tab_domain_diagnostic_dkim_error`);
+      });
+  }
+
+  getTitleDependingOnStepFor(targetService) {
+    return !this.showSpfDiagnosticTitle
+      ? `${targetService}_tab_domain_diagnostic_dkim_title_configuration`
+      : `${targetService}_tab_domain_diagnostic_spf_title`;
+  }
+
+  getNextButtonDependingOnStepFor(targetService) {
+    return this.showConfiguratingBtn
+      ? `${targetService}_tab_domain_diagnostic_dkim_configurate_no_ovhcloud`
+      : `${targetService}_tab_domain_diagnostic_dkim_next_no_ovhcloud`;
+  }
+
+  initSpfContext() {
+    this.showSpfDiagnosticTitle = true;
+    return this.getSelectorNameForNoOvhCloud();
+  }
+
+  getDkimNameFor(targetService, index) {
+    return [
+      this.services.$translate.instant(
+        `${targetService}_tab_domain_diagnostic_dkim_name`,
+      ),
+      `${index}: `,
+      this[`selector${index}NoDomain`].customerRecord,
+    ].join('');
+  }
+
+  getDkimRecordFor(targetService, index) {
+    return [
+      this.services.$translate.instant(
+        `${targetService}_tab_domain_diagnostic_dkim_record`,
+      ),
+      `${index}: `,
+      this[`selector${index}NoDomain`].targetRecord,
+    ].join('');
+  }
+
+  hideConfirmButton() {
+    return (
+      this.dkimStatus === this.DKIM_STATUS.TO_CONFIGURE &&
+      this.domainDiag.isOvhDomain
+    );
+  }
+
+  writeError(msg, args) {
+    this.writeMessage(msg, 'error', args);
+  }
+
+  writeSuccess(msg, args) {
+    this.writeMessage(msg, 'success', args);
+  }
+
+  writeMessage(msg, type, args) {
+    const message = this.services.$translate.instant(msg, args);
+    this.services.$scope.setMessage(message, {
+      type,
+    });
+  }
```

Comment:
**nitpick:** You are only using these functions once. I dont see the need to create 3 additional wrappers for them.
Also, please avoid shortcuts and use fullnames, `msg` -> `message`, `args` -> `arguments` and so on.